### PR TITLE
SPCTR-2567 - Fix: Image Gallery - Migrated Focus List from an Array to an Object

### PR DESCRIPTION
**Pull request by bsf-zanev**
### Description
Created a fixed attribute object to migrate the required focus list elements from the array to remove all null entries.

### Types of changes
Bug fix (non-breaking change which fixes an issue)

### How has this been tested?
Tested as per the comment in the related task.

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Review by RajkiranBagal — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Review by patilvikasj — APPROVED** [review]

File: blocks-config/image-gallery/class-spectra-image-gallery.php [hunk 1]
```diff
@@ -941,6 +945,15 @@ public function render_initial_grid( $attributes ) {
 					}
 				}
 
+				// Check if the new Object Focus List is empty and the old Array Focus List is not - if so, transfer it.
+				if ( empty( $attributes['focusListObject'] ) && is_array( $attributes['focusList'] ) && 0 !== count( $attributes['focusList'] ) ) {
```
Comment (adi3890) [inline blocks-config/image-gallery/class-spectra-image-gallery.php:949]:
```suggestion
				if ( empty( $attributes['focusListObject'] ) && is_array( $attributes['focusList'] ) && ! empty( $attributes['focusList'] ) ) {
```
should we do it this way

File: blocks-config/image-gallery/class-spectra-image-gallery.php [hunk 2]
```diff
@@ -1308,7 +1321,7 @@ private function render_masonry_hover_handler( $mediaArray, $atts ) {
 		private function render_single_media( $mediaArray, $atts ) {
 			// Check if this is part of the Tiled Layout, and if so then check if the current image is focused or not.
 			$focusedClass = '';
-			if ( 'tiled' === $atts['feedLayout'] && ( array_key_exists( $mediaArray['id'], $atts['focusList'] ) && ( true === $atts['focusList'][ $mediaArray['id'] ] ) ) ) {
+			if ( 'tiled' === $atts['feedLayout'] && ( array_key_exists( $mediaArray['id'], $atts['focusListObject'] ) && ( true === $atts['focusListObject'][ $mediaArray['id'] ] ) ) ) {
```
Comment (adi3890) [inline blocks-config/image-gallery/class-spectra-image-gallery.php:1324]:
```suggestion
			if ( 'tiled' === $atts['feedLayout'] && ! empty( $atts['focusListObject'][ $mediaArray['id'] ] ) ) {
```

please check syntax once

File: src/blocks/image-gallery/edit.js [hunk 3]
```diff
@@ -25,6 +39,20 @@ const UAGBImageGallery = ( props ) => {
 		// Assigning block_id in the attribute.
 		setAttributes( { block_id: clientId.substr( 0, 8 ) } );
 		setAttributes( { classMigrate: true } );
+
+		// Replacing the old Focus List Array with the Object List.
+		if ( 'object' === typeof focusList && focusList.length ) {
```
Comment (adi3890) [inline src/blocks/image-gallery/edit.js:44]: can you please confirm if this condition is working.
i have some doubt regarding object.length

Comment (bsf-zanev) [reply src/blocks/image-gallery/edit.js:44]: Yes, this condition works - the typeof for this Array attribute results as an object.

To be more precise though, I've updated this statement to `Array.isArray( focusList )` for the array checks.

File: blocks-config/image-gallery/class-spectra-image-gallery.php (hunk 1 above)
Comment (bsf-zanev) [reply blocks-config/image-gallery/class-spectra-image-gallery.php]: Yes, replaced with Not Empty check.

File: blocks-config/image-gallery/class-spectra-image-gallery.php (hunk 2 above)
Comment (bsf-zanev) [reply blocks-config/image-gallery/class-spectra-image-gallery.php]: Done, replaced with Not Empty check for the current ID's value in the Focus List Object.

**Review by adi3890 — APPROVED** [review]

**Review by supriyajjadhav — APPROVED** [review]

**Pull request merged**